src/app/navigation/vertical-navigation.component.ts

```diff
@@ -0,0 +1,665 @@
+import {
+  Component, ElementRef, EventEmitter,
+  Input, OnDestroy, OnInit, Output, Renderer2, TemplateRef,
+  ViewEncapsulation
+} from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+import { NavigationItemConfig } from './navigation-item-config';
+
+/**
+ * Vertical navigation component
+ */
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'pfng-vertical-navigation',
+  styleUrls: ['./vertical-navigation.component.less'],
+  templateUrl: './vertical-navigation.component.html',
+  host: {
+    '(window:resize)': 'onResize($event)'
+  }
+})
+export class VerticalNavigationComponent implements OnInit, OnDestroy {
+
+  /**
+   * Source for the brand logo at the top
+   */
+  @Input() brandSrc: string;
+
+  /**
+   * Alternate text to display for the logo
+   */
+  @Input() brandAlt: string;
+
+  /**
+   * Boolean to indicate whether or not to show badges
+   */
+  @Input() showBadges: boolean;
+
+  /**
+   * Indicates whether or not to allow the secondary to persist
+   */
+  @Input() persistentSecondary: boolean;
+
+  /**
+   * Allow pinnable menus when they are open
+   */
+  @Input() pinnableMenus: boolean;
+
+  /**
+   * Hide menu items
+   */
+  @Input() hiddenIcons: boolean;
+
+  /**
+   * The navigation items used to build the menu
+   */
+  @Input() items: NavigationItemConfig[];
+
+  /**
+   * Sets an active flag on items when they are selected
+   */
+  @Input() updateActiveItemsOnClick: boolean;
+
+  /**
+   * Indicates whether or not this is a mobile friendly navigation
+   */
+  @Input() ignoreMobile: boolean;
+
+  /**
+   * Hide top banner optionally
+   */
+  @Input() hideTopBanner: boolean;
+
+  /**
+   * This event is fired any time the user has initiated navigation
+   */
+  @Output('onNavigatedEvent') navigatedEvent = new EventEmitter();
```

Comment:
Thinking this should be named onNavigationEvent? Been trying to ovoid past tense, plurals, etc.